packages/blade/src/components/List/_decisions/decisions.md

```diff
@@ -0,0 +1,219 @@
+# List <!-- omit in toc -->
+Lists display a set of related items that are composed of text/links. Each list item begins with a bullet or a number.
+
+<img src="./list-thumbnail.png" width="380" />
+
+- [Design](#design)
+  - [Unordered List](#unordered-list)
+    - [Unordered List with 3 levels of nesting](#unordered-list-with-3-levels-of-nesting)
+    - [Unordered List with custom icons for bullets](#unordered-list-with-custom-icons-for-bullets)
+    - [Unordered List with inline Link in items](#unordered-list-with-inline-link-in-items)
+  - [Ordered List](#ordered-list)
+    - [Ordered List with 3 levels of nesting](#ordered-list-with-3-levels-of-nesting)
+    - [Ordered List with 3 levels of nesting and _filled background_ numbers](#ordered-list-with-3-levels-of-nesting-and-filled-background-numbers)
+    - [Ordered List with inline Link in items](#ordered-list-with-inline-link-in-items)
+- [API](#api)
+  - [List Component](#list-component)
+  - [List.Item Component](#listitem-component)
+  - [List.Item.Link Component](#listitemlink-component)
+  - [Sample Usage](#sample-usage)
+  - [Alternative APIs](#alternative-apis)
+    - [Hierarchial without explicit nested `List` grouping](#hierarchial-without-explicit-nested-list-grouping)
+    - [Hierarchial with a `nested` prop](#hierarchial-with-a-nested-prop)
+    - [Non-Hierarchial with a `level` prop](#non-hierarchial-with-a-level-prop)
+- [Accessibility](#accessibility)
+- [Open Questions](#open-questions)
+- [References](#references)
+
+
+## Design
+- [Figma - List](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=23205%3A446859&t=dT849yOxgX5WHgby-4)
+
+### Unordered List
+#### Unordered List with 3 levels of nesting
+
+<img src="./unordered-list.png" width="380" />
+
+#### Unordered List with custom icons for bullets
+
+<img src="./unordered-list-icon.png" width="380" />
+
+#### Unordered List with inline Link in items
+
+<img src="./unordered-list-link.png" width="380" />
+
+### Ordered List
+#### Ordered List with 3 levels of nesting
+
+<img src="./ordered-list.png" width="380" />
+
+#### Ordered List with 3 levels of nesting and _filled background_ numbers
+
+<img src="./ordered-list-filled.png" width="380" />
+
+#### Ordered List with inline Link in items
+
+<img src="./ordered-list-link.png" width="380" />
+
+
+## API
+
+### List Component
```

Comment:
Alternative: We can have 2 components, `UnorderedList` and `OrderedList`. `OrderedList` can have `variant="filled"`.
```ts
import { UnorderedList, OrderedList } from '@razorpay/blade/components'
```

This will also keep things similar to how they are in HTML.

Reply:
Having it closer to HTML will also help in MDX (which is where I believe lists will be used heavily. We can directly pass our blade components to MDX overrides https://github.com/razorpay/docs/blob/062498c779dacdc8c3c661f043378204e8540237/docs-website/src/shared/components/layouts/mdx/ProductPageLayout.tsx#L49-L54 without having to create wrappers)

Reply:
Although I see some cons would be:
- Can create confusion with 2 component names
- Might not be intuitive to search for "UnorderedList" from components.

Reply:
The major con is discoverability. A single List component is much easier to understand and discover which is why I kept it as a single component. I see most other design systems doing the same except a few. We can still take a call on this with opinions from others.

Reply:
Makes sense. I like the approach Anurag suggested in above comment. Can go with that `order-filled` variant instead and avoid these components.